### PR TITLE
stream: avoid sanitizer warnings for strm->tx

### PR DESCRIPTION
**Pull request by cspiel1**

### CODE REVIEW

**Comment by sreimers** [comment]
Maybe we should postpone after release. RX Thread mode is experimental and this is a big change before release.

**Comment by cspiel1** [comment]
I think it has not much or nothing to do with the RX thread. Also nothing with the TX send thread.

The TX data is driven by the Microphone thread (ausrc thread). And SIP is handled by the main thread. Should we merge it now, or later? The concurrency failure is very old.

**Comment by sreimers** [comment]
> I think it has not much or nothing to do with the RX thread. Also nothing with the TX send thread.

Ah I see, yes let's postpone after release.

File: src/stream.c [hunk 1]
```diff
@@ -1502,7 +1525,14 @@ enum media_type stream_type(const struct stream *strm)
 
 int stream_pt_enc(const struct stream *strm)
 {
-	return strm ? strm->tx.pt_enc : -1;
+	int pt;
+
+
+	mtx_lock(strm->tx.lock);
```
Comment (sreimers) [inline src/stream.c:1531]: `strm` null check is missing here, possible null pointer deref.

File: src/stream.c [hunk 2]
```diff
@@ -1595,7 +1626,6 @@ int stream_debug(struct re_printf *pf, const struct stream *s)
 	err |= re_hprintf(pf, " tx.enabled: %s\n",
 			  re_atomic_rlx(&s->tx.enabled) ? "yes" : "no");
 	err |= rtprecv_debug(pf, s->rx);
```
Comment (sreimers) [inline src/stream.c:1628]: Should we reorder `rtprecv_debug` and `rtp_debug`? So we can unlock earlier? Or is `tx.lock` needed for `rtprecv_debug` too?

**Comment by sreimers** [comment]
Please rebase against main, this should fix:
```
FATAL: ThreadSanitizer: unexpected memory mapping 0x59f14d472000-0x59f14d4a3000
```

**Pull request merged**